lib/hanami/commands/generate/action.rb

```diff
@@ -234,14 +234,16 @@ def spec_root
         end
 
         def relative_action_path
-          path = Pathname.new('.').join('..', '..', '..')
-          path = path.join('..') if environment.container?
+          nestings = ['..'] * @controller_pathname.count('/')
```

Comment:
@berfarah 
> At the top of action.rb, the @controller_pathname variable is assigned by using Hanami::Utils::String().underscore.

Ouch, this should be addressed. The usage of `underscore` is useful to transform `Foo::Bar::Baz` into predictable chunks. But then the mistake was to use the output string directly without to normalize via `::File::Separator`.